docker/driver.go

```diff
@@ -0,0 +1,96 @@
+package docker
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bblfsh/performance"
+
+	errors "gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-log.v1"
+)
+
+// One day it will become a config file but now it's to annoying to ship it
+var drivers = map[string]string{
+	"bash":       "bash-driver",
+	"cpp":        "cpp-driver",
+	"csharp":     "csharp-driver",
+	"go":         "go-driver",
+	"java":       "java-driver",
+	"javascript": "javascript-driver",
+	"php":        "php-driver",
+	"python":     "python-driver",
+	"ruby":       "ruby-driver",
+	"typescript": "typescript-driver",
+}
+
+// InstallDriver downloads driver's repository, checkouts to a given commit,
+// builds docker image and installs driver's image to bblfshd container.
+// Requires bblfshd container running.
+func InstallDriver(language, commit string) error {
+	if err := validateLanguage(language); err != nil {
+		return err
+	}
+
+	driver := drivers[language]
+	image := driverImage(driver, commit)
+	if err := downloadAndBuildDriver(driver, commit, image); err != nil {
+		return err
+	}
+	if err := installDriverToBblfshd(bblfshdContainer, language, image); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func downloadAndBuildDriver(driver, commit, image string) error {
+	tmpDir := filepath.Join(os.TempDir(), driver)
+	log.Debugf("creating directory %v", tmpDir)
+	if err := os.MkdirAll(tmpDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	repo := driverUrl(driver)
+	log.Debugf("performing git clone repository %v %v", repo, commit)
+	cloneCommand := fmt.Sprintf("git clone %s %s ; cd %s ; git checkout %s", repo, tmpDir, tmpDir, commit)
```

Comment:
```go
clone := fmt.Sprintf(`git clone %[1]s %[2]s ; cd %[2]s ; git checkout %[3]s`, repo, tmpDir, commit)
```

Consider maybe also importing https://godoc.org/bitbucket.org/creachadair/shell and using `shell.Quote(repo)`, `shell.Quote(tmpDir)` to ensure the strings are properly protected in the script.

(The same advice applies to the other command lines in this file)

Reply:
Done